src/adam_core/orbit_determination/tests/test_iod.py

```diff
@@ -0,0 +1,108 @@
+import importlib.util
+import os
+import sys
+
+import numpy as np
+import pyarrow.compute as pc
+import pytest
+
+from ...coordinates import CoordinateCovariances, SphericalCoordinates
+from ...coordinates.origin import Origin
+from ...observers import Observers
+from ...utils.helpers.observations import make_observations
+from ...utils.helpers.orbits import make_real_orbits
+from ..evaluate import OrbitDeterminationObservations
+from ..iod import iod
+
+# Specify the path to `adam_assist` in `site-packages`
+site_packages_path = next(
+    (p for p in sys.path if "__pypackages__" in p and p.endswith("lib")), None
+)
+
+try:
+    assist_path = os.path.join(
+        site_packages_path, "adam_core", "propagator", "adam_assist.py"
+    )
+    # Import `adam_assist` from `site-packages`
+    spec = importlib.util.spec_from_file_location(
+        "adam_core.propagator.adam_assist", assist_path
+    )
+    adam_assist = importlib.util.module_from_spec(spec)
+    sys.modules["adam_core.propagator.adam_assist"] = adam_assist
+    spec.loader.exec_module(adam_assist)
+    from adam_core.propagator.adam_assist import ASSISTPropagator  # noqa: E402
+except Exception as e:
+    ASSISTPropagator = None
+    print(e)
```

Comment:
Before we merge this in, I'd like to see if we can get this to work by adding `adam-assist` as an editable dependency using the git repo.

https://pdm-project.org/latest/usage/dependency/#editable-dependencies

This will require changing the way we do test / dev dependencies, away from `optional-dependencies` and instead using the pdm specific method.

https://pdm-project.org/latest/usage/dependency/#add-development-only-dependencies